### PR TITLE
[BottomAppBar] Add readme and Swift example

### PR DESCRIPTION
**Pull request by jgunaratne**
Adds readme file for bottom app bar as well as Swift example.

### CODE REVIEW

File: components/BottomAppBar/README.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  <img src="docs/assets/bottomappbar.png" alt="Text Fields" width="375">
+</div>
+
+A bottom app bar provides a docked bar at the bottom of the screen for common application actions. The bottom app bar includes a floating button for a primary action and a navigation bar area for secondary actions. Transition animations are provided when the floating button shifts left or right, based on the application navigation state, and when the floating action button changes elevation or is hidden.
+
+## Installation
+
+### Requirements
+
+- Xcode 7.0 or higher.
```
Comment (willlarche) [inline components/BottomAppBar/README.md]: Xcode 8 right now.

File: components/BottomAppBar/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+pod install
+~~~
+
+- - -
+
+### Overview
+
+Bottom app bars follow a recommended Material Design interaction design pattern for providing primary and secondary actions that are easily accessible. With a bottom app bar users are more easily able to use single-handed touch interaction with an application since actions are displayed close to the bottom of the screen within easy reach of a user's thumb.
+
+The bottom app bar includes a floating action button that is intended to provide users with a primary action. Secondary actions are available on a navigation bar that can be customized with several buttons on the left and right sides of the navigation bar. The primary action floating action button is centered on the bottom app bar by default.
```
Comment (willlarche) [inline components/BottomAppBar/README.md]: Make `floating action button` a link to the component.

Comment (willlarche) [reply components/BottomAppBar/README.md]: Make all first occurrences of component names links if you can.

File: components/BottomAppBar/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+MDCBottomAppBarView can be added to a view hierarchy like any UIView. Material Design guidelines recommend always placing the bottom app bar at the bottom of the screen.
+
+### Importing
+
+Before using Bottom App Bar, you'll need to import it:
+
+<!--<div class="material-code-render" markdown="1">-->
+#### Swift
+
+~~~ swift
```
Comment (willlarche) [inline components/BottomAppBar/README.md]: Change `~~~` to ` ``` `.

File: components/BottomAppBar/README.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+bottomBarView.floatingButton.addTarget(self,
+                                       action: #selector(didTapFloatingButton(_:)),
+                                       for: .touchUpInside)
+
+// Set the image on the floating button.
+let addImage = UIImage(named:"Add")
+bottomBarView.floatingButton.setImage(addImage, for: .normal)
+
+// Theme the floating button (optional).
+let colorScheme = MDCBasicColorScheme(primaryColor: UIColor.white)
```
Comment (willlarche) [inline components/BottomAppBar/README.md]: Cut `UIColor`. Just have `.white`.

File: components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+import MaterialComponents
+
+class BottomAppBarTypicalUseSwiftExample: UIViewController {
+
+  let bottomBarViewHeight = CGFloat(96);
```
Comment (willlarche) [inline components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift]: Do `let bottomBarViewHeight: CGFloat = 96` instead and drop the `;`.

The current syntax is type casting.

File: components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+    // Set the image on the floating button.
+    let addImage = UIImage(named:"Add")
+    bottomBarView.floatingButton.setImage(addImage, for: .normal)
+
+    // Set the position of the floating button.
+    bottomBarView.floatingButtonPosition = .center
+
+    // Theme the floating button.
+    let colorScheme = MDCBasicColorScheme(primaryColor: UIColor.white)
```
Comment (willlarche) [inline components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift]: `.white`

File: components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+    let barButtonTrailingItem = UIBarButtonItem()
+    let searchImage = UIImage(named:"Search")
+    barButtonTrailingItem.image = searchImage
+
+    bottomBarView.leadingBarButtonItems = [ barButtonLeadingItem ]
+    bottomBarView.trailingBarButtonItems = [ barButtonTrailingItem ]
+  }
+
+  func didTapFloatingButton(_ sender : MDCFloatingButton) {
```
Comment (willlarche) [inline components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift]: For Swift 4 compatibility, add `@objc` to the signature.

File: components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    if (bottomBarView.floatingButtonPosition == .center) {
+      bottomBarView.setFloatingButtonPosition(.trailing, animated: true)
+    } else {
+      bottomBarView.setFloatingButtonPosition(.center, animated: true)
+    }
+  }
+
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    self.view.backgroundColor = UIColor.white
```
Comment (willlarche) [inline components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift]: `.white`

**Review by willlarche — DISMISSED** [review]

File: components/BottomAppBar/README.md (hunk 1 above)
Comment (jgunaratne) [reply components/BottomAppBar/README.md]: Done.

File: components/BottomAppBar/README.md (hunk 2 above)
Comment (jgunaratne) [reply components/BottomAppBar/README.md]: Done.

File: components/BottomAppBar/README.md (hunk 3 above)
Comment (jgunaratne) [reply components/BottomAppBar/README.md]: Done.

File: components/BottomAppBar/README.md (hunk 4 above)
Comment (jgunaratne) [reply components/BottomAppBar/README.md]: Done.

File: components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift (hunk 5 above)
Comment (jgunaratne) [reply components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift]: Done.

File: components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift (hunk 6 above)
Comment (jgunaratne) [reply components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift]: Done.

File: components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift (hunk 7 above)
Comment (jgunaratne) [reply components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift]: Done.

File: components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift (hunk 8 above)
Comment (jgunaratne) [reply components/BottomAppBar/examples/BottomAppBarTypicalUseExample.swift]: Done.

File: components/BottomAppBar/README.md (hunk 4 above)
Comment (ianegordon) [reply components/BottomAppBar/README.md]: This is still UIColor.white.
git push with the updates?

**Review by ianegordon — DISMISSED** [review]

File: components/BottomAppBar/README.md (hunk 4 above)
Comment (jgunaratne) [reply components/BottomAppBar/README.md]: Done.

**Review by ianegordon — APPROVED** [review]

**Pull request merged**